packages/typescript/test/test.js

```diff
@@ -139,39 +139,66 @@ test.serial('supports emitting types also for single file output', async (t) =>
   // as that would have the side effect that the tsconfig's path would be used as fallback path for
   // the here unspecified outputOptions.dir, in which case the original issue wouldn't show.
   process.chdir('fixtures/basic');
+  const outputOpts = { format: 'es', dir: 'dist' };
 
   const warnings = [];
   const bundle = await rollup({
     input: 'main.ts',
+    output: outputOpts,
     plugins: [typescript({ declaration: true, declarationDir: 'dist' })],
     onwarn(warning) {
       warnings.push(warning);
     }
   });
   // generate a single output bundle, in which case, declaration files were not correctly emitted
-  const output = await getCode(bundle, { format: 'es', file: 'dist/main.js' }, true);
+  const output = await getFiles(bundle, outputOpts);
 
   t.deepEqual(
     output.map((out) => out.fileName),
-    ['main.js', 'dist/main.d.ts']
+    ['dist/main.js', 'dist/main.d.ts']
+  );
+  t.is(warnings.length, 0);
+});
+
+test.serial('supports emitting declarations in correct directory for output.file', async (t) => {
+  // Ensure even when no `output.dir` is configured, declarations are emitted to configured `declarationDir`
+  process.chdir('fixtures/basic');
+  const outputOpts = { format: 'es', file: 'dist/main.esm.js' };
```

Comment:
I'd like to also see a test where the output file doesn't match the declaration dir

Reply:
What do you mean? Something like this?:
```
output.file = 'dist/main.js'
declarationDir = 'types'
```

This will always result in `dist/types/main.d.ts`, because rollup does not allow files outside the `output.dir` or the base directory of the `output.file`, see this [rollups source](https://github.com/rollup/rollup/blob/bceb3539bab95ba92688f29a7c9f3e9effc44698/src/rollup/rollup.ts#L268).